File: src/main/java/org/lanternpowered/server/network/vanilla/command/SuggestionType.java
/*
 * Lantern
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) SpongePowered <https://www.spongepowered.org>
 * Copyright (c) contributors
 *
 * This work is licensed under the terms of the MIT License (MIT). For
 * a copy, see 'LICENSE.txt' or <https://opensource.org/licenses/MIT>.
 */
package org.lanternpowered.server.network.vanilla.command;

public final class SuggestionType {

    public static SuggestionType of(String id) {
        return new SuggestionType(id);
    }

    private final String id;

    private SuggestionType(String id) {
        this.id = id;
    }

    public String getId() {
        return this.id;
    }
}

File: src/main/kotlin/org/lanternpowered/server/network/vanilla/command/SuggestionType.kt
/*
 * Lantern
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) SpongePowered <https://www.spongepowered.org>
 * Copyright (c) contributors
 *
 * This work is licensed under the terms of the MIT License (MIT). For
 * a copy, see 'LICENSE.txt' or <https://opensource.org/licenses/MIT>.
 */
package org.lanternpowered.server.network.vanilla.command

class SuggestionType private constructor(val id: String) {

    companion object {

        @JvmStatic
        fun of(id: String): SuggestionType {
            return SuggestionType(id)
        }
    }
}
